CMakeLists.txt

```diff
@@ -32,14 +32,22 @@ cmake_policy(SET CMP0048 NEW)
 
 project(absl CXX)
 
-# when absl is included as subproject (i.e. using add_subdirectory(abseil-cpp))
-# in the source tree of a project that uses it, install rules are disabled.
+# detect if absl is the root CMake project, or is a subproject in the source tree
+# of a project that uses it (e.g. using add_subdirectory(abseil-cpp))
 if(NOT "^${CMAKE_SOURCE_DIR}$" STREQUAL "^${PROJECT_SOURCE_DIR}$")
+  # disable install rules
   set(ABSL_ENABLE_INSTALL FALSE)
+  set(ABSL_IS_CMAKE_SUBPROJECT TRUE)
```

Comment:
Looks like ABSL_ENABLE_INSTALL could just be killed here for logic in ABSL_IS_CMAKE_SUBPROJECT.  IIRC this only serves to make the install CMake rule visible, it doesn't control if Abseil is actually installed anywhere.

Reply:
Sure! Cleaned up `ABSL_ENABLE_INSTALL.` I had to introduce the variable with the complementary boolean value because I could not do an inline negation of the value of `ABSL_ENABLE_INSTALL` when passing the default value for the `ABSL_DECLARE_PUBLIC_SYSTEM_HEADERS` option.